### PR TITLE
Fix lamp color after error or misconfiguration is fixed and oversupply of trains

### PR DESCRIPTION
**Pull request by mrvn**
Trying to get the colors right for my multiple-stops stations patch I noticed that the colors are already wrong in some cases. Looking deeper I noticed it affects not just the colors.

First for the colors. On error the station turns red. But when the error is cleared the station becomes green even if there is a train parked at the station or on the way to the station.
My first fix was to mark the station blue if a train is still parked there. Turning a station yellow if a train is still scheduled to arrive was harder and lead me to the next problem:

Currently whenever there is a misconfiguration of a stop or a stop is a depot you reset the stop.activeDeliveries. That means any knowledge of pending trains is lost. If the problem is cleared LTN will then dispatch new trains to the station. Among others this happens every time you blueprint a station with names included. And sending too many trains causes all kinds of chaos.

### CODE REVIEW

File: script/stop-update.lua [hunk 1]
```diff
@@ -77,22 +78,17 @@ function UpdateStop(stopID)
   stop.providePriority = 0
   stop.lockedSlots = 0
 
-  -- reject any stop not in name list
+  -- readd any stop not in name list
   if not global.TrainStopNames[stop.entity.backer_name] then
```
Comment (Yousei9) [inline script/stop-update.lua]: When a stop is not in global.TrainStopNames something went horribly wrong.
Adding such a stop to normal processing cycle could introduce problems I can't even think of.

**Comment by mrvn** [comment]
I couldn't reproduce it nor can I think of any reason why a stop wouldn't be in global.TrainStopNames. I assumed it was maybe some migration issue. I'm fine with leaving that part out since it never happens anyway.

My reasoning for adding the stop back though was that if a stop is not listed and you create a second stop with the same name then that is listed but the duplicate name detection fails. Then trains get send to the wrong station randomly.

**Comment by Yousei9** [comment]
It might be redundant 99% of the time.
I had cases of stops being removed without giving any event notification.

Good point.
I wonder if several months from now this will solve an issue or cause one.

File: script/stop-update.lua [hunk 2]
```diff
@@ -59,12 +59,13 @@ function UpdateStop(stopID)
     global.LogisticTrainStops[stopID].parkedTrainID = nil
   end
 
-  -- remove invalid activeDeliveries -- shouldn't be necessary
-  -- for i=#stop.activeDeliveries, 1, -1 do
-    -- if not global.Dispatcher.Deliveries[stop.activeDeliveries[i]] then
-      -- table.remove(stop.activeDeliveries, i)
-    -- end
-  -- end
+  -- remove invalid activeDeliveries -- shouldn't be necessary -- only check oldest delivery
```
Comment (Yousei9) [inline script/stop-update.lua]: Do your changes require this?
Why only check oldest? The 3rd of 5 deliveries can become invalid if the train doing it was destroyed.

**Pull request merged**